### PR TITLE
Fixes: #4875 monkeypatch st2 sensor earlier

### PR DESCRIPTION
**Pull request by punkrokk**
Fixes: #4975 
Fixes: #4832

### CODE REVIEW

**Comment by punkrokk** [comment]
@alexku7 Could you test this branch out and tell me if it resolves #4832 for you?

**Comment by guzzijones** [comment]
Yeah monley patching should always be the very first thing. Before imports.

File: st2reactor/st2reactor/container/sensor_wrapper.py [hunk 1]
```diff
@@ -14,6 +14,18 @@
 
 from __future__ import absolute_import
 
+# Note: We need to perform monkey patching in the worker. If we do it in
+# the master process (gunicorn_config.py), it breaks tons of things
+# including shutdown
+# NOTE: It's important that we perform monkey patch as early as possible before any other modules
+# are important, otherwise SSL support for sensor clients won't work.
```
Comment (cognifloyd) [inline st2reactor/st2reactor/container/sensor_wrapper.py:21]:
```suggestion
# are imported, otherwise SSL support for sensor clients won't work.
```

**Comment by guzzijones** [comment]
It looks like these are all in the `cmd` folder so they are command scripts that are run by for services.  This way of monkey patching should work just fine. And yes it needs to be before any other imports.  If we have other cases of monkey patching we need to do the same. 
I don't see any of these scripts used as imports at first glance which is what I thought would be an issue.

**Comment by punkrokk** [comment]
@Kami Do you have any suggestions for how to add a test case for this? 

The only thing I can think of is adding one that fails if we see a recursion error?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -34,13 +34,17 @@ Fixed
...
+  #4832 and https://github.com/gevent/gevent/issues/1016) Contributed by @punkrokk
+
 Removed
 ~~~~~~~
 
 * Removed ``CentOS 6``/``RHEL 6`` support #4984
 
   Contributed by Amanda McGuinness (@amanda11 Ammeon Solutions)
   
+
```
Comment (armab) [inline CHANGELOG.rst:47]:
```suggestion
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -34,13 +34,17 @@ Fixed
 
   Contributed by Nick Maludy (@nmaludy Encore Technologies)
 
+* Fixed a bug where a python3 sensor using ssl needs to be monkey patched earlier. #4975 ( See also
+  #4832 and https://github.com/gevent/gevent/issues/1016) Contributed by @punkrokk
```
Comment (armab) [inline CHANGELOG.rst:38]:
```suggestion
* Fixed a bug where a python3 sensor using ssl needs to be monkey patched earlier. (bug fix) #4976
  See also #4832, #4975 and https://github.com/gevent/gevent/issues/1016
  
  Contributed by @punkrokk
```

**Comment by punkrokk** [comment]
Forum thread: https://forum.stackstorm.com/t/centos-8-problems-with-rabbitmq-connections-when-sensor-dispatches-trigger/1328/12

**Pull request merged**